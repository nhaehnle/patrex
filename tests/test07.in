	if (igbase().get_display_flag(Interactive_Base::dfDebug)) {
		capsbuttons->add
			(new UI::Callback_Button
				(capsbuttons, "debug",
					0, 0, 34, 34,
					g_gr->get_picture(PicMod_UI, "pics/but4.png"),
					g_gr->get_picture(PicMod_Game,  pic_debug),
					boost::bind(&Building_Window::act_debug, boost::ref(*this)),
					_("Debug")),
				UI::Box::AlignCenter);
	}

	capsbuttons->add
		(new UI::Callback_Button
			(capsbuttons, "goto",
				0, 0, 34, 34,
				g_gr->get_picture(PicMod_UI, "pics/but4.png"),
				g_gr->get_picture(PicMod_Game, "pics/menu_goto.png"),
				boost::bind(&Building_Window::clicked_goto, boost::ref(*this))),
			UI::Box::AlignCenter);

	if (m_building.descr().has_help_text()) {
		capsbuttons->add_inf_space();
		capsbuttons->add
			(new UI::Callback_Button
				(capsbuttons, "help",
				0, 0, 34, 34,
				g_gr->get_picture(PicMod_UI, "pics/but4.png"),
				g_gr->get_picture(PicMod_Game, "pics/menu_help.png"),
				boost::bind(&Building_Window::help_clicked, boost::ref(*this)),
				_("Help")),
				UI::Box::AlignCenter);
	}
