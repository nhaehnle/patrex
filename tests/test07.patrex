
define arg $!(,)*
define bind boost::bind($.*)
define new new UI::Callback_Button(${arg}|first|, ${arg}|name|, $(${arg}, ${arg}, ${arg}, ${arg}, ${arg}, ${arg})|mid|, ${bind}|slot| $.*|rest|)

match $<|before| $>|this| capsbuttons->add( ${new}|new|, $.* );
	letrange gap before this
	letregex indent gap "\s*(\n[ \t]*)"
	letregex thename name ".*\"(.*)\".*"
	insert this "UI::Button * {thename}btn ={indent}\tnew UI::Button({first}, {name}, {mid}{rest});{indent}"
	insert this "{thename}btn->sigclicked.connect({slot});{indent}"
	replace new "{thename}btn"
