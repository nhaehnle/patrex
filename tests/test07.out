	if (igbase().get_display_flag(Interactive_Base::dfDebug)) {
		UI::Button * debugbtn =
			new UI::Button(capsbuttons, "debug", 0, 0, 34, 34,
					g_gr->get_picture(PicMod_UI, "pics/but4.png"),
					g_gr->get_picture(PicMod_Game,  pic_debug),
					_("Debug"));
		debugbtn->sigclicked.connect(boost::bind(&Building_Window::act_debug, boost::ref(*this)));
		capsbuttons->add
			(debugbtn,
				UI::Box::AlignCenter);
	}

	UI::Button * gotobtn =
		new UI::Button(capsbuttons, "goto", 0, 0, 34, 34,
				g_gr->get_picture(PicMod_UI, "pics/but4.png"),
				g_gr->get_picture(PicMod_Game, "pics/menu_goto.png"));
	gotobtn->sigclicked.connect(boost::bind(&Building_Window::clicked_goto, boost::ref(*this)));
	capsbuttons->add
		(gotobtn,
			UI::Box::AlignCenter);

	if (m_building.descr().has_help_text()) {
		capsbuttons->add_inf_space();
		UI::Button * helpbtn =
			new UI::Button(capsbuttons, "help", 0, 0, 34, 34,
				g_gr->get_picture(PicMod_UI, "pics/but4.png"),
				g_gr->get_picture(PicMod_Game, "pics/menu_help.png"),
				_("Help"));
		helpbtn->sigclicked.connect(boost::bind(&Building_Window::help_clicked, boost::ref(*this)));
		capsbuttons->add
			(helpbtn,
				UI::Box::AlignCenter);
	}
